src/shared/components/SubmissionPage/FilestackFilePicker/index.jsx

```diff
@@ -113,16 +114,23 @@ class FilestackFilePicker extends React.Component {
       'url',
     ];
 
-    if (communitiesList.data && communitiesList.data.length > 0) {
-      const topGearCommunity = _.find(communitiesList.data, { mainSubdomain: 'topgear' });
-      if (topGearCommunity) {
-        _.forOwn(groups, (value, key) => {
-          if (value && _.includes(topGearCommunity.groupIds, key)) {
-            pickupSources = ['url'];
-            return false;
-          }
-          return true;
-        });
+    // check if challenge belong to any group
+    if (!_.isEmpty(groups)) {
+      // check the group info and match with group list
+      if (communitiesList.data && communitiesList.data.length > 0) {
```

Comment:
I checked reducer behavior, actually `tcCommunities.list.data`, from where `communitiesList.data` comes from, defaults to `[]`, thus the first part of the check makes no sense; and the second one is not correct, because a user can have no groups. The correct check will be to check that `tcCommunities.list.timestamp` is positive, which means that list has been fetched.

Reply:
@birdofpreyru fixed tcCommunities.list.timestamp and styling of URL link button in the latest commit